### DISCUSSION TITLE
Runtime made custom themes - dynamic tailwind.config.js

### DISCUSSION THREAD

**Discussion by lveillard** [tailwindlabs/tailwindcss#8949] [Help] [answered]
Hello guys! I'm just curious to know if this can be done with tailwind or if it makes sense as a feature request.

Let's imagine you create a form builder using tailwind. This form builder can be used by lot's of different companies that want different themes. 

Could it be possible to create a dynamic tailwind.config.js so we get the theme from DB for each client, and we apply the right theming for each one of those clients?

So, an easy example. Let say each client company has a "primary" color. So instead of having a huge tailwind file like this:
- primary-company1: #D5FAF8
- primary-company2: #70EF86
....

we could just have:
`- primary: getPrimary(clientId)`

Thanks!

**Comment by wongjn** [comment, marked as answer]
You could use custom variables for this! Generate the values of the CSS custom variables on the server and print them out. See the [documentation on CSS variable colors](https://tailwindcss.com/docs/customizing-colors#using-css-variables) for more details.

## Example

Generated CSS by the server from the data from the DB:

```css
.client-1-form {
  --color-primary: 200 100 0;
}
.client-2-form {
  --color-primary: 100 200 0;
}
```

`tailwind.config.js` would be something like:

```js
module.exports = {
  theme: {
    extends: {
      colors: {
        primary: {
          DEFAULT: 'rgb(var(--color-primary) / <alpha-value>)',
        },
      },
    },
  },
};
```

And then you can use with classes like `text-primary` or even with opacities like `background-primary/50`.

**Reply by lveillard** [reply]
Hello! Thanks for the answer, i will explore this.

How could I build this:
```
.client-1-form {
  --color-primary: 200 100 0;
}
.client-2-form {
  --color-primary: 100 200 0;
}
```
On runtime? So for example if the DB gets updated with a new color for client-2, i actually get that colour without deploying again the app? 🤔

**Reply by wongjn** [reply]
This is outside Tailwind territory now but you could build the CSS on every relevant request, something like:

```js
const css = db()
  .getRows('clients')
  .map(([id, color]) => `
.client-${id}-form {
  --color-primary: ${color};
}`)
  .join();
```
And then you can serve it how you would like.
